mountpoint-s3/tests/fuse_tests/write_test.rs

```diff
@@ -836,55 +839,6 @@ fn overwrite_fail_on_write_without_truncate_test_mock(write_only: bool) {
     );
 }
 
-fn overwrite_no_truncate_test<F>(creator_fn: F, prefix: &str, write_only: bool)
-where
-    F: FnOnce(&str, TestSessionConfig) -> (TempDir, BackgroundSession, TestClientBox),
-{
-    let filesystem_config = S3FilesystemConfig {
-        allow_overwrite: true,
-        ..Default::default()
-    };
-    let test_config = TestSessionConfig {
-        filesystem_config,
-        ..Default::default()
-    };
-    let (mount_point, _session, mut test_client) = creator_fn(prefix, test_config);
-
-    // Make sure there's an existing directory and a file
-    test_client.put_object("dir/hello.txt", b"hello world").unwrap();
-
-    let _subdir = mount_point.path().join("dir");
-    let path = mount_point.path().join("dir/hello.txt");
-
-    // Open the file in non-truncate mode and do nothing
-    let mut options = File::options();
-    if !write_only {
-        options.read(true);
-    }
-    let write_fh = options.write(true).open(&path).expect("open should succeed");
-    drop(write_fh);
-
-    // File content should not be changed
-    let mut options = File::options();
-    let mut fh = options.read(true).write(true).open(&path).unwrap();
-    let mut hello_contents = String::new();
-    fh.read_to_string(&mut hello_contents).unwrap();
-    assert_eq!(hello_contents, "hello world");
-}
```

Comment:
we do not believe we want to allow this behavior. i.e. open for writing only, closing file, no change.

cc @jamesbornholt